dotnet/packages/Microsoft.Bot.Builder.M365/Microsoft.Bot.Builder.M365/TypingTimer.cs

```diff
@@ -36,56 +40,98 @@ public TypingTimer(int interval = 1000)
         /// the current activity is not a "message" the call is ignored.
         /// </remarks>
         /// <param name="turnContext">The context for the current turn with the user.</param>
-        public void StartTypingTimer(ITurnContext turnContext)
+        /// <returns>True if the timer was started, otherwise False.</returns>
+        public bool Start(ITurnContext turnContext)
         {
-            if (turnContext.Activity.Type != ActivityTypes.Message || timer != null) return;
+            if (turnContext.Activity.Type != ActivityTypes.Message || IsRunning()) return false;
+
+            _timerState = new TimerState(turnContext, false);
 
             // Listen for outgoing activities
             turnContext.OnSendActivities(StopTimerWhenSendMessageActivityHandler);
-            
+
             // Start periodically send "typing" activity
-            timer = new Timer(SendTypingActivity, turnContext, 0, _interval);
-        }
+            _timer = new Timer(SendTypingActivity, _timerState, 0, _interval);
```

Comment:
I believe we can simplify much of this code by using the Change() method inside of the SendTypingActivity callback

```
_timer = new Timer(SendTypingActivity, null, Timeout.Infinite, Timeout.Infinite);
_timer.Change(0,Timeout.Infinite); // fire first time

// inside of callback
private async void SendTypingActivity(object state){
 
  // send and await typing indicator
  // make sure errors are handled if necessary
   await turnContext.SendActivityAsync(new Activity { Type = ActivityTypes.Typing });
  
  // wait 1 sec and fire again
  _timer.Change(1000, Timeout.Infinite);
}

```

Reply:
fixed